service/frontend/workflowHandler.go

```diff
@@ -2758,11 +2753,40 @@ func (wh *WorkflowHandler) QueryWorkflow(
 		response.GetClientFeatureVersion(),
 		response.GetClientImpl(),
 	)
-
 	queryRequest.Execution.RunId = response.Execution.RunId
-	if len(response.StickyTaskList.GetName()) != 0 && clientFeature.SupportStickyQuery() {
-		matchingRequest.TaskList = response.StickyTaskList
-		stickyDecisionTimeout := response.GetStickyTaskListScheduleToStartTimeout()
+	if clientFeature.SupportConsistentQuery() {
```

Comment:
so currently this path is not exercised right?

Reply:
Correct